### DISCUSSION TITLE
BSON Timestamp in a mongoose schema

### DISCUSSION THREAD

**Discussion by blowfishlol** [Automattic/mongoose#14619] [Q&A]
Hello,

Is there a way to have a BSON Timestamp as a mongoose schema? 
https://www.mongodb.com/docs/manual/reference/bson-types/#timestamps

The use case for this is that we are processing several information from mongo's changestream and we need to store the clusterTime given by the events to maintain order of the events.

Thanks in advance 😃

**Comment by vkarpov15** [comment]
Not currently, but we will look into adding that

**Reply by vkarpov15** [reply]
We just published a plugin for this: https://www.npmjs.com/package/@mongoosejs/bson-timestamp

**Reply by blowfishlol** [reply]
Thanks :D Appreciate it. Will take a look into it.
